src/main/java/com/google/cloud/spanner/pgadapter/utils/StatementParser.java

```diff
@@ -141,9 +141,6 @@ private static String removeCommentsAndTrimPostgres(String sql) {
       throw new IllegalArgumentException(
           "SQL statement contains an unterminated block comment: " + sql);
     }
-    if (res.length() > 0 && res.charAt(res.length() - 1) == SEMICOLON) {
```

Comment:
Why do we need this? If I understand correctly this removes the last semicolon that should be safe to do? (and it might be likely to be the correct thing to do)

Reply:
This was removed because the code now removes comments at the query processing phase which means the pre-created queries in our regexes would all need to be changed to remove the ending ';'.  It also doesn't need to be removed so I think it is better to just change the code here.